engines/router/missionctl/log/resultlog/upi_result_log.go

```diff
@@ -48,36 +53,139 @@ var convertorTableSchema uint32 = converter.TableSchemaV1
 
 var routerNameRegex = regexp.MustCompile(`^[a-zA-Z0-9_\-]+-\d+.[a-zA-Z0-9\-]+$`)
 
-func InitUPIResultLogger(appName string, logger UPILogger) (*UPIResultLogger, error) {
+func InitUPIResultLogger(
+	appName string,
+	loggerType config.ResultLogger,
+	upiLogger UPILogger,
+	resultLogger *ResultLogger) (*UPIResultLogger, error) {
 
-	resultLogger := &UPIResultLogger{
-		logger: logger,
+	upiResultLogger := &UPIResultLogger{
+		upiLogger:          upiLogger,
+		turingResultLogger: resultLogger,
+		loggerType:         loggerType,
 	}
 	if !routerNameRegex.MatchString(appName) {
 		return nil, fmt.Errorf("invalid router name")
 	}
 	s := strings.Split(appName, ".")
 	routerNameWithVersion := s[0]
-	resultLogger.projectName = s[1]
+	upiResultLogger.projectName = s[1]
 
 	i := strings.LastIndex(routerNameWithVersion, "-")
-	resultLogger.routerName = routerNameWithVersion[:i]
+	upiResultLogger.routerName = routerNameWithVersion[:i]
 	// do not include '-'
-	resultLogger.routerVersion = routerNameWithVersion[i+1:]
+	upiResultLogger.routerVersion = routerNameWithVersion[i+1:]
 
-	return resultLogger, nil
-}
+	// check the respective logger has been provided for the config
+	switch loggerType {
```

Comment:
nit: I am thinking if it's necessary to check again, `InitUPIResultLogger` is called internally by Turing so all its arguments will likely to be valid, do I miss anything?

Reply:
hmm yes it will actually be valid for all cases and redudant, I can remove it, no opinion on this. 

I was thinking more of a package pov, when a client want to init this result logger, these checks are necessary to ensure valid configuration to make the rest of the methods work and should not assume on the client (application.go in this case) to pass in correct config